libdnf5/rpm/solv/goal_private.cpp

```diff
@@ -243,16 +248,17 @@ bool GoalPrivate::limit_installonly_packages(libdnf5::solv::IdQueue & job, Id ru
     ::Pool * pool = *spool;
     bool reresolve = false;
 
+    // Use maps to handle each package just once
+    libdnf5::solv::SolvMap marked_for_install_providers_map(spool.get_nsolvables());
+    libdnf5::solv::SolvMap available_unused_providers_map(spool.get_nsolvables());
+
     for (int i = 0; i < installonly.size(); ++i) {
```

Comment:
Suggestion: I believe that it would be possible to iterate over using  `for (Id install_only_provide : installonly) {`
It should be more clear in macro `FOR_PROVIDES(p, pp, installonly[i]) {` what is used.